src/lib.rs

```diff
@@ -16,24 +15,62 @@ pub extern "C" fn __c_verify_plonk_bn254(
     cvkeydir: *const core::ffi::c_char,
     cproof: *const core::ffi::c_char,
     cpkey: *const core::ffi::c_char,
-    cparams: *const core::ffi::c_char,
+    ccommitted_values: *const core::ffi::c_char,
 ) -> core::ffi::c_uint
 {
     let vkeydir = unsafe { CStr::from_ptr(cvkeydir) };
     let proof = unsafe { CStr::from_ptr(cproof) };
     let pkey = unsafe { CStr::from_ptr(cpkey) };
-    let params = unsafe { CStr::from_ptr(cparams) };
+    let committed_values = unsafe { CStr::from_ptr(ccommitted_values) };
 
     return verify_plonk_bn254(
         vkeydir.to_str().expect("verifying key directory name is not a valid string"),
         proof.to_str().expect("proof is not a value string"),
         pkey.to_str().expect("program key is not a valid string"),
-        params.to_str().expect("public params is not a valid string"),
+        committed_values.to_str().expect("committed values are not a valid string"),
+    );
+}
+
+#[no_mangle]
+pub extern "C" fn __c_verify_plonk_bn254_hashed(
```

Comment:
The [canonical](https://datatracker.ietf.org/doc/html/rfc8032#section-4) naming suffix for this API would be `_prehashed`, which conveys that hashing is *always* involved, and insists just on *where* it's performed.